readthedocs/projects/views/public.py

```diff
@@ -106,45 +107,53 @@ def get_context_data(self, **kwargs):
         return context
 
 
-def _badge_return(redirect, url):
-    if redirect:
-        return HttpResponseRedirect(url)
+def _badge_return(version, status):
+    template = loader.get_template('badges/badge.svg')
+    if status == 'passing':
```

Comment:
Just a nitpick, but using a `{'passing': ..., 'failing': ..., 'unknown': ...}` dictionary would be simpler and shorter.